File: 38. Count and Say/38. Count and Say.cs
public class Solution {
    public string CountAndSay(int n) {
        if (n == 1) return "1";
        
        string prev = CountAndSay(n - 1);
        StringBuilder result = new StringBuilder();
        int count = 1;
        
        for (int i = 1; i <= prev.Length; i++) {
            if (i < prev.Length && prev[i] == prev[i-1]) {
                count++;
            } else {
                result.Append(count.ToString() + prev[i-1]);
                count = 1;
            }
        }
        
        return result.ToString();
    }
}

File: 38. Count and Say/38. Count and Say.dart
class Solution {
  String countAndSay(int n) {
    if (n == 1) return "1";
    
    String prev = countAndSay(n - 1);
    StringBuffer result = StringBuffer();
    int count = 1;
    
    for (int i = 1; i <= prev.length; i++) {
      if (i < prev.length && prev[i] == prev[i-1]) {
        count++;
      } else {
        result.write('$count${prev[i-1]}');
        count = 1;
      }
    }
    
    return result.toString();
  }
}